ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+ToolbarDelegate.swift

```diff
@@ -416,138 +416,150 @@ extension BrowserViewController: TopToolbarDelegate {
+      .compactMap({ $0.browserViewController })
 
-      viewController.dismiss(animated: true) {
-        self?.shredData(for: url, in: tab)
-        self?.dismiss(animated: true)
+    browsers.forEach { browser in
+      // Update the shields status immediately
+      browser.topToolbar.refreshShieldsStatus()
+
+      // Reload the tabs. This will also trigger an update of the brave icon in `TabLocationView` if
+      // the setting changed is the global `.AllOff` shield
+      browser.tabManager.allTabs.forEach {
+        if $0.url?.baseDomain == currentDomain {
+          $0.reload()
+        }
       }
     }
+  }
 
-    shields.showGlobalShieldsSettings = { [unowned self] vc in
-      vc.dismiss(animated: true) {
-        weak var spinner: SpinnerView?
-        let controller = UIHostingController(
-          rootView: AdvancedShieldsSettingsView(
-            settings: AdvancedShieldsSettings(
-              profile: self.profile,
-              tabManager: self.tabManager,
-              feedDataSource: self.feedDataSource,
-              historyAPI: self.braveCore.historyAPI,
-              p3aUtilities: self.braveCore.p3aUtils,
-              deAmpPrefs: self.braveCore.deAmpPrefs,
-              debounceService: DebounceServiceFactory.get(privateMode: false),
-              clearDataCallback: { [weak self] isLoading, isHistoryCleared in
-                guard let self else { return }
-                guard let view = self.navigationController?.view, view.window != nil else {
-                  assertionFailure()
-                  return
-                }
-
-                if isLoading, spinner == nil {
-                  let newSpinner = SpinnerView()
-                  newSpinner.present(on: view)
-                  spinner = newSpinner
-                } else {
-                  spinner?.dismiss()
-                  spinner = nil
-                }
-
-                if isHistoryCleared {
-                  // Donate Clear Browser History for suggestions
-                  let clearBrowserHistoryActivity = ActivityShortcutManager.shared
-                    .createShortcutActivity(type: .clearBrowsingHistory)
-                  self.userActivity = clearBrowserHistoryActivity
-                  clearBrowserHistoryActivity.becomeCurrent()
-                }
-              }
-            )
-          )
-        )
+  private func showGlobalShieldsSettings() {
+    weak var spinner: SpinnerView?
+    let controller = UIHostingController(
+      rootView: AdvancedShieldsSettingsView(
+        settings: AdvancedShieldsSettings(
+          profile: self.profile,
+          tabManager: self.tabManager,
+          feedDataSource: self.feedDataSource,
+          historyAPI: self.braveCore.historyAPI,
+          p3aUtilities: self.braveCore.p3aUtils,
+          deAmpPrefs: self.braveCore.deAmpPrefs,
+          debounceService: DebounceServiceFactory.get(privateMode: false),
```

Comment:
`DebounceServiceFactory` doesn't require a `braveCore` instance?

Reply:
Nope, it's a singleton factory class as other examples that use these factories